File: app/src/main/java/com/lcl6/cn/basedialog/mvp/presenter/ManagerPresenter.java
package com.lcl6.cn.basedialog.mvp.presenter;

import com.lcl6.cn.basedialog.mvp.contract.ManagerContract;
import com.lcl6.cn.basedialog.mvp.model.ManagerModel;
import com.lcl6.cn.basedialog.mvp.model.impl.ManagerModelImpl;
import com.lcl6.cn.component.base.mvp.presnenter.RxPresenter;

/**
 * Created by liancl on 2017/9/1.
 */

public class ManagerPresenter extends RxPresenter<ManagerContract.View> implements ManagerContract.Presenter {
    ManagerModelImpl managerModel;
    public ManagerPresenter() {
        managerModel = new ManagerModelImpl();
    }

    @Override
    public void requestTabaoData() {
        managerModel.requestTabaoData(new ManagerModel.LoadCompleteListener() {
            @Override
            public void onComplete(String successMessage) {
                mView.requestTabaoData(successMessage);
            }
            @Override
            public void onFail(String errorMessage) {
                mView.requestTabaoData(errorMessage);
            }
        });
    }

    @Override
    public void setGlobal(String newUrl) {
        managerModel.setGlobal(new ManagerModel.LoadCompleteListener() {
            @Override
            public void onComplete(String successMessage) {
                mView.setGlobal(successMessage);
            }

            @Override
            public void onFail(String errorMessage) {
                mView.setGlobal(errorMessage);
            }
        },newUrl);
    }

    @Override
    public void removeGlobal() {
        managerModel.requestTabaoData(new ManagerModel.LoadCompleteListener() {
            @Override
            public void onComplete(String successMessage) {
                mView.requestTabaoData(successMessage);
            }

            @Override
            public void onFail(String errorMessage) {
                mView.requestTabaoData(errorMessage);
            }
        });
    }

    @Override
    public void requestGuthub(String newUrl) {
        managerModel.requestGuthub(new ManagerModel.LoadCompleteListener() {
            @Override
            public void onComplete(String successMessage) {
                mView.requestGuthub(successMessage);
            }

            @Override
            public void onFail(String errorMessage) {
                mView.requestGuthub(errorMessage);
            }
        },newUrl);
    }

    @Override
    public void requestGank(String newUrl) {
        managerModel.requestGank(new ManagerModel.LoadCompleteListener() {
            @Override
            public void onComplete(String successMessage) {
                mView.requestGank(successMessage);
            }
            @Override
            public void onFail(String errorMessage) {
                mView.requestGank(errorMessage);
            }
        },newUrl);
    }

    @Override
    public void requestDouban(String newUrl) {
        managerModel.requestDouban(new ManagerModel.LoadCompleteListener() {
            @Override
            public void onComplete(String successMessage) {
                mView.requestDouban(successMessage);
            }
            @Override
            public void onFail(String errorMessage) {
                mView.requestDouban(errorMessage);
            }
        },newUrl);
    }
}

File: app/src/main/java/com/lcl6/cn/basedialog/mvp/presenter/ManagerPresenter.kt
package com.lcl6.cn.basedialog.mvp.presenter

import com.lcl6.cn.basedialog.mvp.contract.ManagerContract
import com.lcl6.cn.basedialog.mvp.model.ManagerModel
import com.lcl6.cn.basedialog.mvp.model.impl.ManagerModelImpl
import com.lcl6.cn.component.base.mvp.presnenter.RxPresenter

/**
 * Created by liancl on 2017/9/1.
 */

class ManagerPresenter : RxPresenter<ManagerContract.View>(), ManagerContract.Presenter {
    //    @Inject
    internal var managerModel: ManagerModelImpl

    init {
        this.managerModel = ManagerModelImpl()
    }

    override fun requestTabaoData() {
        managerModel.requestTabaoData(object : ManagerModel.LoadCompleteListener {
            override fun onComplete(successMessage: String) {
                mView.requestTabaoData(successMessage)
            }

            override fun onFail(errorMessage: String) {
                mView.requestTabaoData(errorMessage)
            }
        })
    }

    override fun setGlobal(newUrl: String) {
        managerModel.setGlobal(object : ManagerModel.LoadCompleteListener {
            override fun onComplete(successMessage: String) {
                mView.setGlobal(successMessage)
            }

            override fun onFail(errorMessage: String) {
                mView.setGlobal(errorMessage)
            }
        }, newUrl)
    }

    override fun removeGlobal() {
        managerModel.requestTabaoData(object : ManagerModel.LoadCompleteListener {
            override fun onComplete(successMessage: String) {
                mView.requestTabaoData(successMessage)
            }

            override fun onFail(errorMessage: String) {
                mView.requestTabaoData(errorMessage)
            }
        })
    }

    override fun requestGuthub(newUrl: String) {
        managerModel.requestGuthub(object : ManagerModel.LoadCompleteListener {
            override fun onComplete(successMessage: String) {
                mView.requestGuthub(successMessage)
            }

            override fun onFail(errorMessage: String) {
                mView.requestGuthub(errorMessage)
            }
        }, newUrl)
    }

    override fun requestGank(newUrl: String) {
        managerModel.requestGank(object : ManagerModel.LoadCompleteListener {
            override fun onComplete(successMessage: String) {
                mView.requestGank(successMessage)
            }

            override fun onFail(errorMessage: String) {
                mView.requestGank(errorMessage)
            }
        }, newUrl)
    }

    override fun requestDouban(newUrl: String) {
        managerModel.requestDouban(object : ManagerModel.LoadCompleteListener {
            override fun onComplete(successMessage: String) {
                mView.requestDouban(successMessage)
            }

            override fun onFail(errorMessage: String) {
                mView.requestDouban(errorMessage)
            }
        }, newUrl)
    }
}
